securedrop_client/gui/__init__.py

```diff
@@ -16,193 +16,11 @@
 You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from typing import Union
-
-from PyQt5.QtCore import QSize, Qt
-from PyQt5.QtWidgets import QHBoxLayout, QLabel, QPushButton, QWidget
-
-# Allow the buttons to be imported directly.
+# Expose GUI elements without exposing the package structure
 from securedrop_client.gui.buttons import SDPushButton  # noqa: F401
```

Comment:
why have an unused import here? The comment to expose gui elements without exposing package structure may confuse future devs. I think we should just remove this line.

Reply:
This import is what allows to use `SDPushButton` from `securedrop_client.gui`, the `__init__.py` file is the file that defines the API of the package. In this case, as far as a consumer is concerned, the package defines an `SDPushButton` class and the fact that the package is internally split into multiple files is an implementation detail.

None of the imports present in this file will be used _in the file itself_, but they're used in other files, and the linter is overzealous in this case.

You can look at the `__init__.py` files in #1369 and #1374 for further examples of how the API of `securedrop_client.gui` is built from its internal packages. 

The goal is to distinguish between what's the public API of each package and what is private. Eventually imports shouldn't need to be _deeply_ nested (`from securedrop_client.gui import SourceList` vs `from securedrop_client.gui.source.list import List`) because the consumer shouldn't be coupled to the internal structure of a package. (It is the package version of [Demeter's guideline][wiki]: too many "dots" in an import are a smell.)

  [wiki]: https://en.m.wikipedia.org/wiki/Law_of_Demeter

I introduced relative import paths in #1374, that you'll see help making the distinction visually between imports that _define_ APIs (in `__init__.py_ files, should all be relative) and consumer imports which happen outside of `__init__.py` files and refer to sibling packages (e.g. when `securedrop_client.logic` is referenced from `securedrop_client.gui`). Currently `securedrop_client.gui` etc. is just repeated everywhere and obscures the package hierarchy a bit.

That's a bit of a long explanation, am I making sense?

Reply:
The comment was attempting to explain why the import was not removed, even though the linter would complain that it was (apparently) not used! That didn't work! 😝

Reply:
Okay, the change makes sense. Let's update this comment `Expose GUI elements without exposing the package structure` to something more prescriptive, like, "Put imports you want accessible from securedrop_client.gui here."